### ISSUE TITLE
Music metadata not showing

### ISSUE THREAD

**Issue by feross** [webtorrent/webtorrent-desktop#1833]
**What version of WebTorrent Desktop are you using?** `master` (soon to be 0.22.0)

**What operating system and version?** macOS

**What happened?**

In 0.21.0, this is what it looked like:

![play-torrent-wired-2](https://user-images.githubusercontent.com/121766/87610016-00081300-c6b9-11ea-98a8-72ee41b2de5b.png)

In 0.22.0, this is what it looks like:

![play-torrent-wired-2-failed](https://user-images.githubusercontent.com/121766/87610041-0eeec580-c6b9-11ea-8327-641cf5c499a2.png)

**What did you expect to happen?**

Music metadata should show

**Are you willing to submit a pull request to fix this bug?**

cc @Borewit

**Comment by Borewit** [comment]
I just wanted to report the issue myself, missed you cc, I will have a look.

The following error appears in the server process:
```
error getting audio metadata for 1802c566f743f5100fb3a01ed8728c082d375093:0 Error: An object could not be cloned.
    at EventEmitter.i.send.i.send (electron/js2c/renderer_init.js:74)
    at Object.observer (C:\Users\Borewit\cod…r\webtorrent.js:346)
    at MetadataCollector.setFormat (C:\Users\Borewit\cod…dataCollector.js:61)
    at MP4Parser.handleAtom (C:\Users\Borewit\cod…p4\MP4Parser.js:205)
    at async Function.readAtom (C:\Users\Borewit\cod…\lib\mp4\Atom.js:25)
    at async MP4Parser.parse (C:\Users\Borewit\cod…p4\MP4Parser.js:119)
    at async Function._parse (C:\Users\Borewit\cod…arserFactory.js:158)
    at async Object.parseFile (C:\Users\Borewit\cod…ata\lib\index.js:44)
```

**Comment by Borewit** [comment]
Problems is that the observable used to retrieve the metadata, close to real-time, is passing the inter [MetadataCollector object](https://github.com/Borewit/music-metadata/blob/master/lib/common/MetadataCollector.ts) which implements the metadata interface. Problem is in this scenario, to pass the data from main to browser, the data is serialized. The solution is simple, just pass the metadata only, which does not contain functions. 

I used this function btw to detect functions:
```js
function clone(x) {
  return JSON.stringify(x, (key, value) => {
    if (typeof value === 'function') {
      throw new Error('metadata shall not contain functions');
    }
    return value;
  });
}
```
Using something like `clone(metadata)` to understand what the issue was.

I was in doubt where to put the change, in [music-metadata](https://github.com/Borewit/music-metadata) or [webtorrent](https://github.com/webtorrent). The draw back of putting it in [music-metadata](https://github.com/Borewit/music-metadata) is that the internal object needs be converted to a _clean_ every event. Most application won't notice that they get a bit more then strictly specified in the interface,

**Comment by hicom150** [comment]
Fixed with #1847 🎉
